osquery/core/posix/process.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <vector>
+
+#include <signal.h>
+
+#include <sys/types.h>
+
+#include <osquery/logger.h>
+
+#include "osquery/core/process.h"
+
+extern char **environ;
+
+namespace osquery {
+
+PlatformProcess::PlatformProcess(PlatformPidType id)
+  : id_(id) { }
+
+bool PlatformProcess::operator==(const PlatformProcess& process) const {
+  return (nativeHandle() == process.nativeHandle());
+}
+
+bool PlatformProcess::operator!=(const PlatformProcess& process) const {
+  return (nativeHandle() != process.nativeHandle());
+}
+
+PlatformProcess::~PlatformProcess() {}
+
+int PlatformProcess::pid() const {
+  return id_;
+}
+
+bool PlatformProcess::kill() const {
+  if (id_ == kInvalidPid) {
+    return false;
+  }
+
+  int status = ::kill(id_, SIGKILL);
+  return (status == 0);
+}
+
+std::shared_ptr<PlatformProcess> PlatformProcess::getCurrentProcess() {
+  pid_t pid = ::getpid();
+  return std::make_shared<PlatformProcess>(pid);
+}
+
+std::shared_ptr<PlatformProcess> PlatformProcess::getLauncherProcess() {
+  pid_t ppid = ::getppid();
+  return std::make_shared<PlatformProcess>(ppid);
+}
+
+std::shared_ptr<PlatformProcess> PlatformProcess::launchWorker(const std::string& exec_path, const std::string& name) {
+  auto worker_pid = ::fork();
+  if (worker_pid < 0) {
+    return std::shared_ptr<PlatformProcess>();
+  } else if (worker_pid == 0) {
+    setEnvVar("OSQUERY_WORKER", std::to_string(::getpid()).c_str());
+    ::execle(exec_path.c_str(), name.c_str(), nullptr, ::environ);
+    
```

Comment:
There's whitespace here, this is one of the things I dislike about Github PRs vs email patches. :/